__tests__/secret.test.js

```diff
@@ -193,6 +193,53 @@ describe('Tozny identity client', () => {
       ops.updateSecret(realmConfig, identity, oldSecret, newSecret)
     ).rejects.toThrow()
   })
+  it('gets the latest version of a secret', async () => {
+    const testName = `test-secret-${uuidv4()}`
+    const oldSecret = {
+      secretType: 'Credential',
+      secretName: testName,
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    const newSecret = {
+      secretType: 'Credential',
+      secretName: testName,
+      secretValue: 'updatedSecretValue',
+      description: 'this is a description',
+    }
+    await ops.createSecret(realmConfig, identity, oldSecret)
+    await ops.updateSecret(realmConfig, identity, oldSecret, newSecret)
+    const start = new Date()
+    let latestVersionOfSecret
+    while (new Date() - start < 30000) {
```

Comment:
I see now why the waitForNext method would not work here... The functional programmer in me says there is a pattern we can abstract out to be even more general, but let's hold off until we see another area that is needed :)

Good job adapting the retry loop for your needs!

Reply:
Lol I had to resist the urge to @ you on this and ask for how to do this with currying, I know it's possible in js